### PR TITLE
[Keyboard] Add Contender

### PR DESCRIPTION
**Pull request by sotoba**
## Description

Adds support for the Contender NumPad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/contender/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/contender/rules.mk:16]:
```suggestion
```

File: keyboards/contender/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/contender/rules.mk:21]:
```suggestion
```

File: keyboards/contender/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u2
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/contender/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/contender/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# contender
+
+<img src="https://github.com/sotoba/contender/blob/images/contender_topview.jpg" width="860">
```
Comment (fauxpark) [inline keyboards/contender/readme.md:3]:
```suggestion
![](https://raw.githubusercontent.com/sotoba/contender/images/contender_topview.jpg)
```

File: keyboards/contender/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+A number pad for hardboot.
+
+* Keyboard Maintainer: [sotoba](https://github.com/sotoba)
+* Hardware Supported: *The contender PCBs, atmega32u2 supported*
```
Comment (fauxpark) [inline keyboards/contender/readme.md:8]:
```suggestion
* Hardware Supported: Contender PCB
```

File: keyboards/contender/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE = 0,
+    _UNRULY = 1,
+    _FUNCTION = 2
```
Comment (fauxpark) [inline keyboards/contender/keymaps/default/keymap.c:22]:
```suggestion
    _BASE,
    _UNRULY,
    _FUNCTION
```

File: keyboards/contender/info.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
+    "keyboard_name": "contender",
+    "url": "https://github.com/sotoba/contender",
+    "maintainer": "sotoba",
+    "width": 5,
+    "height": 6.25,
```
Comment (fauxpark) [inline keyboards/contender/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/contender/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+![](https://raw.githubusercontent.com/sotoba/contender/images/contender_topview.jpg)
+
+A number pad for hardboot.
+
+* Keyboard Maintainer: [sotoba](https://github.com/sotoba)
+* Hardware Supported: Contender PCB
+* Hardware Availability: [BOOTH: stupa-devices](https://stupa-devices.booth.pm/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/contender/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/contender/readme.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# contender
+
+![](https://raw.githubusercontent.com/sotoba/contender/images/contender_topview.jpg)
```
Comment (drashna) [inline keyboards/contender/readme.md:3]: Please rehost this on an image hosting site, such as imgur.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/contender/readme.md (hunk 8 above)
Comment (sotoba) [reply keyboards/contender/readme.md:11]: Add Bootloader section.

https://github.com/qmk/qmk_firmware/pull/15403/files#diff-2b191cee3aa81d3a83a62fe182bf3938ab7cad04d7c2d3c654b1c8d9a60fd853R21

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/contender/config.h [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+#define RGBLED_NUM 33
+#define RGBLIGHT_LED_MAP { \
+   0, 1, 2, 3, \
+   7, 6, 5, 4, \
+   8, 9, 10, 11, \
+   15, 14, 13, 12, \
+   16, 17, 18, 19, \
+   22, 21, 20, \
+   25, 24, 23, 26, 27, 28, 29, 30, 31, 32 }
+#define RGBLIGHT_LAYERS
```
Comment (zvecr) [inline keyboards/contender/config.h:75]:
```suggestion
```

File: keyboards/contender/config.h [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+#define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+#define RGBLIGHT_EFFECT_BREATHING
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
```
Comment (zvecr) [inline keyboards/contender/config.h:89]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
`RGBLIGHT_LAYERS` should be enabled at the keymap level, within `keyboards/contender/keymaps/default/config.h`

**Comment by sotoba** [comment]
> `RGBLIGHT_LAYERS` should be enabled at the keymap level, within `keyboards/contender/keymaps/default/config.h`

fix it. https://github.com/qmk/qmk_firmware/pull/15403/commits/4ea6d7f94ec88d1ec39e7bdb6c0a3e7d508c64b8

**Pull request merged**